pkg/types/v1/parameter.go

```diff
@@ -122,6 +123,19 @@ func (p *Parameter) Validate() error {
 		}
 	case ParameterTypeListstring:
 		_, correctType = p.Value.([]string)
+		if !correctType {
```

Comment:
Covers also from dealing with `[]interface{}` as it happens when we unmarshal `component.yaml` into a Parameter actually 🦀

Reply:
how would a YAML fragment look that would trigger this? I can see a situation with JSON where this could happen but it would mean that the user has added numbers in the list but in YAML it looks like everything is considered a string: https://go.dev/play/p/btjnSip-iKk

also, how would it work that a list is of type []any but it has strings in it? and I don't via a test where you can force the type, but can you replicate this situation with a YAML or a JSON marshaler?

Reply:
Took this offline - ended up implementing a custom YAML marshaler/unmarshaler like we did for JSON for parameters. Tested and now works as expected even with a component.yaml